src/components/liveSession/components/liveSessionBtn/liveSessionBtn.component.js

```diff
@@ -8,52 +8,105 @@
+                let SESSION_DURATION =  {
+                    marginBeforeSessionStart: ENV.liveSession.marginBeforeSessionStart,
+                    marginAfterSessionStart: ENV.liveSession.marginAfterSessionStart
+                };
+                let dataPromMap = {
+                    liveSessionDuration: ZnkLessonNotesSrv.getLiveSessionDuration(),
+                    educatorId: UserProfileService.getCurrUserId()
+                };
 
                 this.$onInit = function() {
                     vm.isLiveSessionActive = false;
                     vm.isOffline = true;
+                    vm.isDiagnosticCompleted = false;
                     vm.endSession = endSession;
-                    vm.showSessionModal = showSessionModal;
+                    vm.showStartSessionPopup = showStartSessionPopup;
                     initializeLiveSessionStatus();
 
-                    $scope.$watch('vm.student', function (newStudent) {
+                    $scope.$watch('vm.student', newStudent => {
                         if (newStudent && angular.isDefined(newStudent.presence)) {
                             vm.isOffline = newStudent.presence === PresenceService.userStatus.OFFLINE;
+                            StudentService.getStudentResults(newStudent.uid).then(studentResults => {
+                                StudentService.isDiagnosticCompleted(studentResults.examResults)
+                                    .then(isDiagnosticCompleted => vm.isDiagnosticCompleted = isDiagnosticCompleted);
+                            });
                         }
                     }, true);
 
                     LiveSessionSrv.registerToCurrUserLiveSessionStateChanges(liveSessionStateChanged);
                 };
 
                 function initializeLiveSessionStatus() {
-                    LiveSessionSrv.getActiveLiveSessionData().then(function (liveSessionData) {
+                    LiveSessionSrv.getActiveLiveSessionData().then(liveSessionData => {
                         if (liveSessionData) {
                             liveSessionStateChanged(liveSessionData.status);
                         }
                     });
                 }
 
-                function showSessionModal() {
-                    $mdDialog.show({
-                        template: '<live-session-subject-modal student="vm.student"></live-session-subject-modal>',
-                        scope: $scope,
-                        preserveScope: true,
-                        clickOutsideToClose: true
+                function showStartSessionPopup() {
+                    if (!vm.isDiagnosticCompleted) {
+                        $log.debug('showStartSessionPopup: Student didn\'t complete Diagnostic test');
+                        return LiveSessionUiSrv.showIncompleteDiagnostic(vm.student.name);
+                    }
+
+                    LiveSessionUiSrv.showStartSessionPopUp().then(() => endSession());
```

Comment:
Where do you close this popup?

Reply:
In liveSessionEvents.service.js when the status changed to CONFIRMED, I close all popups in both educator and student